### PR TITLE
decoder update

### PR DESCRIPTION
**Pull request by Don-Yin**
1. added a deeper decoder
2. added a new eval script for evaluations where the result is store in a json file with the image names as label
3. fixed an issue with multiple workers in dataloader in some specific machine by adding __name__ == "__main__" before the script
4. ignores what comes after "_" in the image file name when as labels.

### CODE REVIEW

File: src/utils/decoder/eval.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import multiprocessing as mp
+from src.utils.net_utils import make_cuda
+from src.utils.misc import imshow_batch
+from tqdm import tqdm
+import json
+import torch
+import os
+
+
+def cprint(text, color="yellow"):
```
Comment (ValerioB88) [inline src/utils/decoder/eval.py:22]: Do we need this? MindSet uses sty for color coding the output
